{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "4a92afa7",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "2db53b72",
   "metadata": {},
   "outputs": [],
   "source": [
    "CODES = [\n",
    "         'AccurateSeg', \n",
    "         'MajorSeg', \n",
    "         'MinorSeg', \n",
    "         'AccurateSection', \n",
    "         'MajorSection', \n",
    "         'MinorSection', \n",
    "         'AccurateEnt', \n",
    "         'MajorEnt',\n",
    "         'MinorEnt'\n",
    "        ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "a6c585ab",
   "metadata": {},
   "outputs": [],
   "source": [
    "present_codes = {}\n",
    "\n",
    "for filename in  os.listdir('AccuracyNew'):\n",
    "    if '.' not in filename:\n",
    "        for file in os.listdir(f\"AccuracyNew/{filename}\"):\n",
    "            if file.endswith('ann'):\n",
    "                with open(os.path.join('AccuracyNew', filename, file)) as f:\n",
    "                    content = f.readlines()\n",
    "                    a_index = 0\n",
    "                    new_lines_to_add = []\n",
    "                    old_lines_to_add = []\n",
    "                    for c in content:\n",
    "                        l = c.split('\\t')\n",
    "                        if l[1].split()[0] in CODES:\n",
    "#                             print(l[1].split()[0])\n",
    "                            curr_code = l[1].split()[0]\n",
    "                            new_code = None\n",
    "                            if curr_code.startswith(\"Accurate\"):\n",
    "                                new_code = \"ACCURATE\"\n",
    "                            elif curr_code.startswith(\"Minor\"):\n",
    "                                new_code = \"Minor\"\n",
    "                            elif curr_code.startswith(\"Major\"):\n",
    "                                new_code = \"Major\"\n",
    "                            \n",
    "                            new_part = l[1].split()\n",
    "                            new_part[0] = new_code\n",
    "                            old_lines_to_add.append(\"\\t\".join([l[0], \" \".join(new_part), l[2]]))\n",
    "      \n",
    "                                \n",
    "                            if curr_code.endswith(\"Ent\"):\n",
    "                                s = \" \".join([\"Level\", l[0], 'Article'])\n",
    "                                nl = '\\t'.join([f\"A{a_index}\", s])\n",
    "                                a_index += 1\n",
    "                                new_lines_to_add.append(nl)\n",
    "\n",
    "                            elif curr_code.endswith(\"Section\"):\n",
    "                                s = \" \".join([\"Level\", l[0], 'Section'])\n",
    "                                nl = '\\t'.join([f\"A{a_index}\", s])\n",
    "                                a_index += 1\n",
    "                                new_lines_to_add.append(nl)\n",
    "                        else:\n",
    "                            old_lines_to_add.append(c)\n",
    "\n",
    "                                \n",
    "                    with open(os.path.join('AccuracyNew', filename, file), 'w') as f:\n",
    "                        f.writelines(old_lines_to_add)\n",
    "                        f.writelines(new_lines_to_add)\n",
    "                                \n",
    "\n",
    "                            \n",
    "                            \n",
    "                            "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "4a6e1f17",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['T3\\tMultiCitation 1192 1196\\t[13]\\n',\n",
       " 'T8\\tContext 824 1209\\tEight of these genes have some evidence in support of an association with autism (http://autismkb.cbi.pku.edu.cn/) [32], two of which (SYN1, PTS) met thresholds for a high level of evidence [32], and concordant changes in the differential expression in autism vs control were found for three of the genes (GPR56, HSPB1, BEX5) between this study and the work of others [13], [28], [33].\\n',\n",
       " 'T1\\tACCURATE 1455 1535\\tTranscriptomic Analysis of Autistic Brain Reveals Convergent Molecular Pathology\\n']"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "old_lines_to_add"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "04a1519f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['A0\\tLevel T1 Article']"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_lines_to_add"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "90e3be74",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['T2', 'AccurateSeg 3951 4165', 'Since disease-causing mutations in PINK1 or Parkin disrupt this pathway, patients with these mutations may not be able to clean up their damaged mitochondria, leading to the neuronal damage typical of parkinsonism.\\n']\n",
      "['T3', 'AccurateSeg 3791 3950', 'In this pathway, PINK1 acts as a flag that accumulates on dysfunctional mitochondria and then signals to Parkin, which tags these mitochondria for destruction.\\n']\n",
      "['T1', 'AccurateSeg 3682 3841', 'In this pathway, PINK1 acts as a flag that accumulates on dysfunctional mitochondria and then signals to Parkin, which tags these mitochondria for destruction.\\n']\n",
      "['T2', 'AccurateSeg 25377 25505', 'A kinase-deficient (KD) version of PINK1 [28] also failed to reconstitute Parkin recruitment to mitochondria (Figure 3A and 3B).\\n']\n",
      "['T2', 'AccurateSeg 52605 52893', 'Although expression of Parkin R275W was moderately less than wild type, it also increases localization in the membrane fraction and decreases in the supernatant upon CCCP treatment, consistent with the mitochondrial translocation seen for this mutation by confocal microscopy (Figure 9C).\\n']\n",
      "['T3', 'AccurateSeg 52044 52216', 'Although under control conditions some mutants formed visible aggregates (Figure S8D), no mutant, including R275W, colocalized with mitochondria (Figure 9B and Figure S8E).\\n']\n",
      "['T4', 'AccurateSeg 54076 54305', 'Interestingly, the R275W mutation in RING1 also exhibited a severe deficit in mitophagy (4.0±4%) (Figure 9D and 9E) even though it appears to largely retain its ability to translocate to uncoupled mitochondria (Figure 9A and 9B).\\n']\n",
      "['T1', 'AccurateEnt 1288 1523', 'PINK1 accumulation on mitochondria is both necessary and sufficient for Parkin recruitment to mitochondria, and disease- causing mutations in PINK1 and Parkin disrupt Parkin recruitment and Parkin- induced mitophagy at distinct steps. \\n']\n",
      "['T1', 'AccurateSeg 33050 33327', 'Consistent with genetic studies in Drosophila, these findings show that Parkin translocation to depolarized mitochondria and Parkin-induced mitophagy are downstream of PINK1 expression, whereas PINK1 accumulation in response to depolarization is upstream of Parkin recruitment.\\n']\n",
      "['T1', 'AccurateSeg 40562 40742', 'Cells treated with vehicle or 250 nM AP21967 for 8 h were scored for mCherry-Parkin in puncta characteristic of mitochondria in ≥150 cells in three or more independent experiments.\\n']\n",
      "['T2', 'MinorEnt 2264 2339', 'PINK1 Is Selectively Stabilized on Impaired Mitochondria to Activate Parkin\\n']\n",
      "['T3', 'MajorSeg 20275 20526', 'When MG132 is added during the final hour of treatment in HeLa cells treated with vehicle, an ∼52-kDa band appears, consistent with the cleavage product of full-length Parkin described in previous reports [22],[28],[30] (Figure 2A, lane 1 vs. lane 2).\\n']\n",
      "['T2', 'AccurateSeg 9713 9957', 'When HeLa cells are treated with CCCP, which depolarizes mitochondria by increasing membrane permeability to H+, a large increase in endogenous full-length PINK1 (∼63 kDa) is seen beginning by 30 min and continuing for at least 3 h (Figure 1A).\\n']\n",
      "['T1', 'AccurateSeg 43554 43864', 'Although these findings do not prove new PINK1 synthesis is required for Parkin recruitment (cycloheximide blocks de novo synthesis of all proteins and thus may inhibit Parkin recruitment independently of PINK1 accumulation), they suggest that PINK1 accumulation and Parkin recruitment may be casually related.\\n']\n",
      "['T2', 'AccurateSeg 42869 43041', 'A 90-min treatment with cycloheximide, likewise, blocked Parkin recruitment to depolarized mitochondria by confocal microscopy (96.0±3.5% vs. 11.3±4.2%) (Figure 7C and 7D).\\n']\n",
      "['T1', 'AccurateSeg 12048 12220', 'To verify that the ∼63-kDa band is in fact PINK1, we immunoblotted for endogenous PINK1 in M17 cells stably transduced with control short hairpin RNA (shRNA) or PINK shRNA.\\n']\n",
      "['T1', 'AccurateSeg 2525 2600', 'PINK1 Is Selectively Stabilized on Impaired Mitochondria to Activate Parkin\\n']\n",
      "['T1', 'AccurateSeg 3944 4208', 'Previous studies in flies and mice, and in human cells suggest that PINK1 and Parkin are part of a common pathway that protects against damaged mitochondria; these organelles power the cell when healthy but can produce harmful reactive oxygen species when damaged.\\n']\n",
      "['T1', 'AccurateSeg 11331 11503', 'To verify that the ∼63-kDa band is in fact PINK1, we immunoblotted for endogenous PINK1 in M17 cells stably transduced with control short hairpin RNA (shRNA) or PINK shRNA.\\n']\n",
      "['T2', 'AccurateSeg 6925 7119', 'In this model, mitochondrial PINK1 is rapidly turned over on bioenergetically well-coupled mitochondria by proteolysis, but is selectively stabilized on mitochondria with low membrane potential.\\n']\n",
      "['T4', 'AccurateEnt 896 971', 'PINK1 Is Selectively Stabilized on Impaired Mitochondria to Activate Parkin\\n']\n",
      "['T1', 'AccurateSeg 22316 22527', \"Consistent with results from a recent study of PINK1's topology when it is ectopically expressed [22], we found that the kinase domain of endogenous PINK1 faces the cytosol following depolarization (Figure S1E).\\n\"]\n",
      "['T3', 'AccurateSeg 33549 33644', 'These data suggest that PINK1 promotes mitophagy in the context of endogenous levels of Parkin.\\n']\n",
      "['T4', 'AccurateSeg 2089 2189', 'Parkin is selectively recruited from the cytosol to damaged mitochondria to trigger their autophagy.\\n']\n",
      "['T5', 'AccurateSeg 57163 57466', 'Our results suggest that full-length mitochondrial PINK1 is the active form in the PINK1/Parkin pathway, and that cleavage of PINK1 into an unstable cytosolic form maintains low levels of PINK1 on healthy mitochondria in order to suppress the PINK1/Parkin pathway in the absence of mitochondrial damage.\\n']\n",
      "['T6', 'AccurateSeg 29774 29976', 'Parkin-dependent mitophagy is reconstituted by exogenous PINK1 expression in the PINK−/− MEFs, with 65.5±5.0% of reconstituted PINK−/− cells displaying undetectable mitochondria following CCCP treatment\\n']\n",
      "['T2', 'AccurateSeg 2255 2524', 'Here, we show that expression of PINK1 on individual mitochondria is regulated by voltage- dependent proteolysis to maintain low levels of PINK1 on healthy, polarized mitochondria, while facilitating the rapid accumulation of PINK1 on mitochondria that sustain damage. \\n']\n",
      "['T3', 'AccurateSeg 7902 8099', 'In this model, mitochondrial PINK1 is rapidly turned over on bioenergetically well-coupled mitochondria by proteolysis, but is selectively stabilized on mitochondria with low membrane potential. Se\\n']\n",
      "['T4', 'AccurateSeg 38324 38511', 'However, OPA3-PINK1 Δ1-110-YFP, which does not display voltage dependent proteolysis, recruits mCherry-Parkin to mitochondria in 98±1.8% of cells in the absence of CCCP (Figure 6D and 6E)\\n']\n",
      "['T5', 'AccurateSeg 55929 56155', 'This selective accumulation is achieved by a novel mechanism, in which PINK1 is constitutively synthesized and imported into all mitochondria, but cleaved from healthy mitochondria by voltage-sensitive proteolysis (Figure S9).\\n']\n",
      "['T6', 'AccurateSeg 8988 9105', 'Remarkably, levels of endogenous mitochondrial PINK1 respond robustly to changes in mitochondrial membrane potential.\\n']\n",
      "['T1', 'AccurateSeg 5031 5157', 'Recent studies have linked Parkin and PINK1 in a pathway critical for the maintenance of mitochondrial integrity and function.\\n']\n",
      "['T3', 'AccurateSeg 986 1111', 'Loss-of-function mutations in PINK1 and Parkin cause parkinsonism in humans and mitochondrial dysfunction in model organisms.\\n']\n",
      "['T1', 'AccurateSeg 54515 54896', 'We recently reported that the Parkinson disease-linked E3 ubiquitin ligase, Parkin, is selectively recruited to dysfunctional mitochondria with low membrane potential to promote their autophagic degradation, suggesting that a deficiency of mitochondrial quality control may underlie the observed mitochondrial dysfunction in Parkin knockout Drosophila and mice [11]–[15],[17],[21].\\n']\n",
      "['T2', 'AccurateSeg 1726 1961', 'PINK1 accumulation on mitochondria is both necessary and sufficient for Parkin recruitment to mitochondria, and disease- causing mutations in PINK1 and Parkin disrupt Parkin recruitment and Parkin- induced mitophagy at distinct steps. \\n']\n",
      "['T3', 'AccurateSeg 3067 3226', 'In this pathway, PINK1 acts as a flag that accumulates on dysfunctional mitochondria and then signals to Parkin, which tags these mitochondria for destruction.\\n']\n",
      "['T4', 'AccurateSeg 3227 3441', 'Since disease-causing mutations in PINK1 or Parkin disrupt this pathway, patients with these mutations may not be able to clean up their damaged mitochondria, leading to the neuronal damage typical of parkinsonism.\\n']\n",
      "['T5', 'AccurateSeg 7451 7590', 'In this model, PINK1 and Parkin form a pathway for sensing and selectively eliminating damaged mitochondria from the mitochondrial network.\\n']\n",
      "['T1', 'AccurateSeg 16559 16755', 'Full- length PINK1 (∼63 kDa), which is anchored in the mitochondrial membrane, is proteolytically cleaved into an ∼52-kDa cytosolic fragment that can be degraded by the proteasome [22],[28]–[30]. \\n']\n",
      "['T2', 'AccurateSeg 19069 19320', 'When MG132 is added during the final hour of treatment in HeLa cells treated with vehicle, an ∼52-kDa band appears, consistent with the cleavage product of full-length Parkin described in previous reports [22],[28],[30] (Figure 2A, lane 1 vs. lane 2).\\n']\n",
      "['T3', 'AccurateSeg 19996 20285', 'Taken together, these results support a two-step model for the processing of PINK1: first, full-length PINK1 is cleaved into the ∼52-kDa short form in a voltage-dependent, proteasome-independent manner, and second, the short form of PINK1 is rapidly degraded by the proteasome (Figure 2B).\\n']\n",
      "['T5', 'AccurateSeg 56050 56347', 'Full-length mitochondrial PINK1 (∼63 kDa) is cleaved into a short ∼52-kDa form, but the short, primarily cytosolic form is unstable, raising the questions: why is PINK1 found both on the mitochondria and in the cytosol, and which form of PINK1 is active in the PINK1/Parkin pathway [22],[28]–[30].\\n']\n",
      "['T6', 'AccurateSeg 57292 57517', 'Although the intramembrane serine protease Rhomboid-7 appears to be required for PINK1 cleavage in Drosophila [31], our results suggest that PARL, its mammalian ortholog, is not required for PINK1 cleavage in mammalian cells.\\n']\n",
      "['T1', 'AccurateSeg 23219 23356', 'Although endogenous PARL could not be detected in HeLa cells, PARL shRNA inhibited expression of overexpressed PARL (Figure S2A and S2B).\\n']\n",
      "['T2', 'AccurateSeg 12743 12950', 'We found that the ∼63-kDa band increases following CCCP treatment in control shRNA cells, but does not increase in the PINK1 shRNA cells, demonstrating that this ∼63-kDa band is endogenous PINK1 (Figure 1B).\\n']\n",
      "['T4', 'AccurateSeg 24244 24412', 'Endogenous PINK1 accumulates similarly in HeLa cells, which display little or no endogenous Parkin expression, and HeLa cells stably expressing YFP-Parkin (Figure S2D).\\n']\n",
      "['T5', 'AccurateSeg 33735 33830', 'These data suggest that PINK1 promotes mitophagy in the context of endogenous levels of Parkin.\\n']\n",
      "['T6', 'AccurateSeg 54288 54551', 'As we found previously [21], expression of wild-type Parkin in HeLa cells that do not detectably express endogenous Parkin completely eliminates mitochondria in greater than half of the cells (59.0±15.1%) following treatment with CCCP for 24 h (Figure 9D and 9E).\\n']\n",
      "['T1', 'AccurateSeg 9323 9567', 'When HeLa cells are treated with CCCP, which depolarizes mitochondria by increasing membrane permeability to H+, a large increase in endogenous full-length PINK1 (∼63 kDa) is seen beginning by 30 min and continuing for at least 3 h (Figure 1A).\\n']\n",
      "['T3', 'AccurateSeg 9569 9777', 'his ∼63-kDa band also increases in the mitochondria-rich membrane fraction following treatment with valinomycin, which, unlike CCCP, depolarizes mitochondria by permeabilizing the membrane to K+ (Figure S1A).\\n']\n",
      "['T1', 'AccurateSeg 23087 23230', 'Likewise, PINK1-V5 levels were similar in PARL−/− and PARL+/+ MEFs, under basal conditions and following depolarization with CCCP (Figure S2C).\\n']\n",
      "['T2', 'AccurateSeg 23231 23307', 'Together, these results suggest that PARL is dispensable for PINK1 cleavage.\\n']\n",
      "['T1', 'AccurateSeg 13225 13510', 'We found that, similar to YFP-Parkin [21], PINK1-YFP accumulates selectively on mitochondria with low membrane potential, demonstrating that PINK1 is selectively stabilized on the depolarized mitochondria within a bioenergetically diverse population of mitochondria (Figure 1E and 1F).\\n']\n",
      "['T2', 'AccurateEnt 7406 7699', \"Here, we show that full-length PINK1 accumulates selectively on dysfunctional mitochondria, and that Parkin recruitment to depolarized mitochondria and subsequent Parkin-induced mitophagy are strictly dependent on PINK1's mitochondrial targeting signal and depolarization-induced accumulation.\\n\"]\n",
      "['T1', 'AccurateSeg 3340 3574', 'PINK1 accumulation on mitochondria is both necessary and sufficient for Parkin recruitment to mitochondria, and disease- causing mutations in PINK1 and Parkin disrupt Parkin recruitment and Parkin- induced mitophagy at distinct steps.\\n']\n",
      "['T3', 'AccurateSeg 7159 7371', 'Interestingly, overexpression of Parkin can partially compensate for PINK1 loss, but PINK1 overexpression cannot compensate for Parkin loss, suggesting that PINK1 functions upstream of Parkin in a common pathway.\\n']\n",
      "['T4', 'AccurateSeg 25193 25345', 'Together, these results show that PINK1 accumulation is upstream of Parkin recruitment to depolarized mitochondria and independent of Parkin expression.\\n']\n",
      "['T1', 'AccurateSeg 64674 64963', 'Although the proximate cause of these abnormalities in PINK1 null cells remains obscure, one explanation may be the failure of PINK1/Parkin pathway to eliminate oxidatively damaged mitochondria, which accumulate over time as a natural consequence of metabolism and other cellular stresses.\\n']\n",
      "['T2', 'AccurateEnt 3142 3356', 'Since disease-causing mutations in PINK1 or Parkin disrupt this pathway, patients with these mutations may not be able to clean up their damaged mitochondria, leading to the neuronal damage typical of parkinsonism.\\n']\n",
      "['T3', 'AccurateSeg 30691 30809', 'We found that Parkin-induced mitophagy is also dependent on PINK1 expression in the M17 human neuroblastoma cell line.\\n']\n",
      "['T4', 'AccurateSeg 30995 31122', 'These results suggest that PINK1 is necessary for the mitophagy of depolarized mitochondria following overexpression of Parkin.\\n']\n",
      "['T6', 'AccurateSeg 62915 63249', 'Alternatively, Parkin may serve other functions in the cell that are independent of PINK1 and protect against mitochondrial dysfunction indirectly; or Parkin may function to some degree upon overexpression independently of mitochondrial docking, perhaps effecting mitophagy or other mitochondrial changes from the cytosolic compartmen\\n']\n",
      "['T1', 'AccurateEnt 5220 5455', 'PINK1 accumulation on mitochondria is both necessary and sufficient for Parkin recruitment to mitochondria, and disease- causing mutations in PINK1 and Parkin disrupt Parkin recruitment and Parkin- induced mitophagy at distinct steps. \\n']\n",
      "['T1', 'AccurateEnt 1099 1174', 'PINK1 Is Selectively Stabilized on Impaired Mitochondria to Activate Parkin\\n']\n",
      "['T1', 'AccurateSeg 32021 32186', 'These results are consistent with endogenous PINK1 promoting mitochondrial degradation in the context of continued (or increased) mitochondrial biogenesis [37],[38].\\n']\n",
      "['T2', 'AccurateSeg 32670 32765', 'These data suggest that PINK1 promotes mitophagy in the context of endogenous levels of Parkin.\\n']\n",
      "['T4', 'AccurateSeg 32766 33009', 'Additionally, these results suggest that the selective turnover of dysfunctional mitochondria may be balanced by the biogenesis of new mitochondria, allowing exchange of damaged, dysfunctional mitochondria for healthy, functional mitochondria.\\n']\n",
      "['T1', 'AccurateSeg 19330 19596', 'In contrast, no differences in LC3-II/β-actin ratios were observed between control- and rapamycin-treated non-transgenic littermates ( Fig. 3a–b ), suggesting that rapamycin may induce autophagy as a response to high Aβ levels in hippocampi of transgenic PDAPP mice.\\n']\n",
      "['T3', 'AccurateSeg 19122 19329', 'In agreement with the expected induction of autophagy by rapamycin-mediated inhibition of mTOR, LC3-II/β-actin ratios in hippocampi of rapamycin-treated PDAPP mice were significantly decreased ( Fig. 3a–b ).\\n']\n",
      "['T2', 'AccurateSeg 1646 1775', \"Inhibition of mTOR by Rapamycin Abolishes Cognitive Deficits and Reduces Amyloid-β Levels in a Mouse Model of Alzheimer's Disease\\n\"]\n",
      "['T1', 'AccurateSeg 20082 20333', 'Our data suggest that the reduction in Aβ42 levels and the improvement in cognitive function in rapamycin-treated PDAPP mice may be a consequence of the induction of autophagy in hippocampus ( Fig. 3 ) by high levels of Aβ in PDAPP transgenic brains. \\n']\n",
      "['T1', 'MajorEnt 859 988', \"Inhibition of mTOR by Rapamycin Abolishes Cognitive Deficits and Reduces Amyloid-β Levels in a Mouse Model of Alzheimer's Disease\\n\"]\n",
      "['T1', 'AccurateSeg 21558 21774', 'In addition, rapamycin treatment did not induce autophagy and did not affect levels of endogenous Aβ in non-transgenic mice, suggesting that autophagy may have a key role in reducing Aβ42 in transgenic PDAPP brains. \\n']\n",
      "['T3', 'AccurateSeg 2775 3085', 'As expected from the inhibition of mTOR, autophagy was increased in neurons of rapamycin-treated transgenic, but not in non-transgenic, PDAPP mice, suggesting that the reduction in Aβ and the improvement in cognitive function are due in part to increased autophagy, possibly as a response to high levels of Aβ.\\n']\n",
      "['T2', 'AccurateSeg 2322 2632', 'As expected from the inhibition of mTOR, autophagy was increased in neurons of rapamycin-treated transgenic, but not in non-transgenic, PDAPP mice, suggesting that the reduction in Aβ and the improvement in cognitive function are due in part to increased autophagy, possibly as a response to high levels of Aβ.\\n']\n",
      "['T3', 'AccurateSeg 20175 20366', 'Taken together, our results suggest that autophagy is induced by rapamycin-mediated mTOR inhibition specifically as a response to high Aβ levels in hippocampi of rapamycin-treated PDAPP mice.\\n']\n",
      "['T1', 'AccurateSeg 1940 2069', \"Inhibition of mTOR by Rapamycin Abolishes Cognitive Deficits and Reduces Amyloid-β Levels in a Mouse Model of Alzheimer's Disease\\n\"]\n",
      "['T1', 'AccurateSeg 1174 1303', \"Inhibition of mTOR by Rapamycin Abolishes Cognitive Deficits and Reduces Amyloid-β Levels in a Mouse Model of Alzheimer's Disease\\n\"]\n",
      "['T2', 'AccurateSeg 22162 22501', 'Although the differences in performance between control-fed and rapamycin-fed non- transgenic groups were not significant, they may suggest that changes in pathways different from autophagy (such as effects on the regulation of protein synthesis) as a result of long-term mTOR inhibition may have a positive effect on learning and memory. \\n']\n",
      "['T3', 'AccurateSeg 1104 1233', \"Inhibition of mTOR by Rapamycin Abolishes Cognitive Deficits and Reduces Amyloid-β Levels in a Mouse Model of Alzheimer's Disease\\n\"]\n",
      "['T1', 'AccurateSeg 20820 21103', 'Consistent with a key role for high levels of Aβ in the activation of autophagy when mTOR activity is reduced, rapamycin did not induce autophagy in brains of rapamycin-treated non-transgenic mice, in which levels of endogenous Aβ are much lower than those in PDAPP transgenic brains\\n']\n",
      "['T2', 'AccurateSeg 1174 1303', \"Inhibition of mTOR by Rapamycin Abolishes Cognitive Deficits and Reduces Amyloid-β Levels in a Mouse Model of Alzheimer's Disease\\n\"]\n",
      "['T2', 'AccurateSeg 3160 3338', 'Our data suggest that inhibition of mTOR by rapamycin, an intervention that extends lifespan in mice, can slow or block AD progression in a transgenic mouse model of the disease.\\n']\n",
      "['T3', 'AccurateSeg 3339 3458', 'Rapamycin, already used in clinical settings, may be a potentially effective therapeutic agent for the treatment of AD.\\n']\n",
      "['T1', 'AccurateSeg 7581 7820', 'Thus, our data suggest that inhibition of mTOR by rapamycin, an intervention that extends lifespan in mice, can lower Aβ levels and slow or block AD progression in a transgenic mouse model of the disease, possibly by stimulating autophagy.\\n']\n",
      "['T1', 'AccurateSeg 22304 22495', 'Taken together, our results suggest that autophagy is induced by rapamycin-mediated mTOR inhibition specifically as a response to high Aβ levels in hippocampi of rapamycin-treated PDAPP mice.\\n']\n",
      "['T1', 'AccurateSeg 2916 3094', 'Our data suggest that inhibition of mTOR by rapamycin, an intervention that extends lifespan in mice, can slow or block AD progression in a transgenic mouse model of the disease.\\n']\n",
      "['T1', 'AccurateSeg 1559 1688', \"Inhibition of mTOR by Rapamycin Abolishes Cognitive Deficits and Reduces Amyloid-β Levels in a Mouse Model of Alzheimer's Disease\\n\"]\n",
      "['T3', 'AccurateSeg 23173 23374', 'In summary, our data suggest that inhibition of mTOR by rapamycin[5], an intervention that extends lifespan in mice[5], [6], can slow or block AD progression in a transgenic mouse model of the disease.\\n']\n",
      "['T4', 'AccurateSeg 3049 3227', 'Our data suggest that inhibition of mTOR by rapamycin, an intervention that extends lifespan in mice, can slow or block AD progression in a transgenic mouse model of the disease.\\n']\n",
      "['T2', 'AccurateSeg 3593 3903', 'As expected from the inhibition of mTOR, autophagy was increased in neurons of rapamycin-treated transgenic, but not in non-transgenic, PDAPP mice, suggesting that the reduction in Aβ and the improvement in cognitive function are due in part to increased autophagy, possibly as a response to high levels of Aβ.\\n']\n",
      "['T4', 'AccurateSeg 24059 24260', 'In summary, our data suggest that inhibition of mTOR by rapamycin[5], an intervention that extends lifespan in mice[5], [6], can slow or block AD progression in a transgenic mouse model of the disease.\\n']\n",
      "['T1', 'AccurateSeg 1465 1594', \"Inhibition of mTOR by Rapamycin Abolishes Cognitive Deficits and Reduces Amyloid-β Levels in a Mouse Model of Alzheimer's Disease\\n\"]\n",
      "['T1', 'AccurateSeg 2057 2366', 'As expected from the inhibition of mTOR, autophagy was increased in neurons of rapamycin-treated transgenic, but not in non-transgenic, PDAPP mice, suggesting that the reduction in Aβ and the improvement in cognitive function are due in part to increased autophagy, possibly as a response to high levels of Aβ\\n']\n",
      "['T1', 'AccurateSeg 1034 1163', \"Inhibition of mTOR by Rapamycin Abolishes Cognitive Deficits and Reduces Amyloid-β Levels in a Mouse Model of Alzheimer's Disease\\n\"]\n",
      "['T3', 'AccurateSeg 22648 22849', 'In summary, our data suggest that inhibition of mTOR by rapamycin[5], an intervention that extends lifespan in mice[5], [6], can slow or block AD progression in a transgenic mouse model of the disease.\\n']\n",
      "['T2', 'AccurateSeg 1465 1594', \"Inhibition of mTOR by Rapamycin Abolishes Cognitive Deficits and Reduces Amyloid-β Levels in a Mouse Model of Alzheimer's Disease\\n\"]\n",
      "['T2', 'AccurateSeg 6632 6783', 'These data indicate that inhibition of the mTOR pathway by long-term rapamycin treatment can reduce Aβ42 levels in vivo and block or delay AD in mice. \\n']\n",
      "['T1', 'AccurateSeg 1117 1246', \"Inhibition of mTOR by Rapamycin Abolishes Cognitive Deficits and Reduces Amyloid-β Levels in a Mouse Model of Alzheimer's Disease\\n\"]\n",
      "['T2', 'MajorSeg 15521 15775', 'The reduction in Aβ42 in PDAPP brains did not likely arise from changes in production, since the abundance of C99, the C-terminal product of β-secretase cleavage of APP, as well the levels of expression of hAPP were unchanged by rapamycin ( Fig. 2f–h ). \\n']\n",
      "['T1', 'AccurateSeg 567 696', \"Inhibition of mTOR by Rapamycin Abolishes Cognitive Deficits and Reduces Amyloid-β Levels in a Mouse Model of Alzheimer's Disease\\n\"]\n",
      "['T2', 'AccurateSeg 2057 2235', 'Our data suggest that inhibition of mTOR by rapamycin, an intervention that extends lifespan in mice, can slow or block AD progression in a transgenic mouse model of the disease.\\n']\n",
      "['T1', 'MajorSeg 22021 22360', 'Although the differences in performance between control-fed and rapamycin-fed non- transgenic groups were not significant, they may suggest that changes in pathways different from autophagy (such as effects on the regulation of protein synthesis) as a result of long-term mTOR inhibition may have a positive effect on learning and memory. \\n']\n",
      "['T1', 'AccurateSeg 805 934', \"Inhibition of mTOR by Rapamycin Abolishes Cognitive Deficits and Reduces Amyloid-β Levels in a Mouse Model of Alzheimer's Disease\\n\"]\n",
      "['T2', 'AccurateSeg 2295 2473', 'Our data suggest that inhibition of mTOR by rapamycin, an intervention that extends lifespan in mice, can slow or block AD progression in a transgenic mouse model of the disease.\\n']\n",
      "['T3', 'AccurateSeg 22419 22620', 'In summary, our data suggest that inhibition of mTOR by rapamycin[5], an intervention that extends lifespan in mice[5], [6], can slow or block AD progression in a transgenic mouse model of the disease.\\n']\n",
      "['T1', 'AccurateSeg 2940 3118', 'Our data suggest that inhibition of mTOR by rapamycin, an intervention that extends lifespan in mice, can slow or block AD progression in a transgenic mouse model of the disease.\\n']\n",
      "['T1', 'AccurateSeg 1727 1856', \"Inhibition of mTOR by Rapamycin Abolishes Cognitive Deficits and Reduces Amyloid-β Levels in a Mouse Model of Alzheimer's Disease\\n\"]\n",
      "['T2', 'AccurateSeg 2757 2875', 'These data indicate that inhibition of the mTOR pathway can reduce Aβ42 levels in vivo and block or delay AD in mice. \\n']\n",
      "['T2', 'AccurateSeg 655 784', \"Inhibition of mTOR by Rapamycin Abolishes Cognitive Deficits and Reduces Amyloid-β Levels in a Mouse Model of Alzheimer's Disease\\n\"]\n",
      "['T3', 'AccurateSeg 2145 2323', 'Our data suggest that inhibition of mTOR by rapamycin, an intervention that extends lifespan in mice, can slow or block AD progression in a transgenic mouse model of the disease.\\n']\n",
      "['T5', 'AccurateSeg 22269 22470', 'In summary, our data suggest that inhibition of mTOR by rapamycin[5], an intervention that extends lifespan in mice[5], [6], can slow or block AD progression in a transgenic mouse model of the disease.\\n']\n",
      "['T1', 'AccurateSeg 683 811', \"nhibition of mTOR by Rapamycin Abolishes Cognitive Deficits and Reduces Amyloid-β Levels in a Mouse Model of Alzheimer's Disease\\n\"]\n",
      "['T3', 'AccurateSeg 22296 22497', 'In summary, our data suggest that inhibition of mTOR by rapamycin[5], an intervention that extends lifespan in mice[5], [6], can slow or block AD progression in a transgenic mouse model of the disease.\\n']\n",
      "['T4', 'AccurateSeg 2172 2350', 'Our data suggest that inhibition of mTOR by rapamycin, an intervention that extends lifespan in mice, can slow or block AD progression in a transgenic mouse model of the disease.\\n']\n",
      "['T1', 'AccurateSeg 15927 16219', 'In addition, the reduction in Aβ42 did not likely arise from increased degradation, since levels of the two major Aβ-degrading activities, neprilysin (NEP)[35], and insulin-degrading enzyme (IDE)[38] were unchanged in brains of control- and rapamycin-fed PDAPP mice as well ( Fig. 2f, j–k ). \\n']\n",
      "['T2', 'AccurateSeg 22820 23159', 'Although the differences in performance between control-fed and rapamycin-fed non- transgenic groups were not significant, they may suggest that changes in pathways different from autophagy (such as effects on the regulation of protein synthesis) as a result of long-term mTOR inhibition may have a positive effect on learning and memory. \\n']\n",
      "['T1', 'AccurateEnt 1124 1243', 'Large-scale association analysis provides insights into the genetic architecture and pathophysiology of type 2 diabetes\\n']\n",
      "['T1', 'AccurateEnt 889 1008', 'Large-scale association analysis provides insights into the genetic architecture and pathophysiology of type 2 diabetes\\n']\n",
      "['T1', 'AccurateEnt 1206 1325', 'Large-scale association analysis provides insights into the genetic architecture and pathophysiology of type 2 diabetes\\n']\n",
      "['T2', 'AccurateSeg 27084 27298', 'We used the published MAGIC meta-analysis (up to 37,037 non-diabetic individuals) of HOMA indices of beta-cell function and insulin sensitivity5 as these traits were not included in the enlarged Metabochip study34.\\n']\n",
      "['T1', 'AccurateEnt 1206 1325', 'Large-scale association analysis provides insights into the genetic architecture and pathophysiology of type 2 diabetes\\n']\n",
      "['T2', 'AccurateSeg 25513 25725', 'To gain a more complete picture of patterns of trait overlap, we first assessed the effect of T2D risk alleles on glycemic traits in European-descent meta- analyses from the MAGIC Investigators (Online Methods). \\n']\n",
      "['T2', 'MinorSeg 10825 11135', 'Figure 1 presents the distribution of Z-scores from Stage 2, aligned to the risk allele from Stage 1, at a subset of 3,412 independent (CEU r2 < 0.05) T2D replication variants that excludes lead SNPs and possible proxies (CEU r2 ≥ 0.1) at the 63 newly discovered and established loci represented on Metabochip.\\n']\n",
      "['T3', 'MinorSeg 12945 13217', 'Using simulations, based on parameter estimates from this mixture model, we estimated that 488 (95% confidence interval (CI) 456–521) of the independent replication SNPs, in addition to the 63 newly discovered and established loci, are associated with T2D susceptibility. \\n']\n",
      "['T2', 'MajorEnt 1469 1588', 'Large-scale association analysis provides insights into the genetic architecture and pathophysiology of type 2 diabetes\\n']\n",
      "['T1', 'MinorEnt 1831 1950', 'Large-scale association analysis provides insights into the genetic architecture and pathophysiology of type 2 diabetes\\n']\n",
      "['T3', 'MinorSeg 11187 11497', 'Figure 1 presents the distribution of Z-scores from Stage 2, aligned to the risk allele from Stage 1, at a subset of 3,412 independent (CEU r2 < 0.05) T2D replication variants that excludes lead SNPs and possible proxies (CEU r2 ≥ 0.1) at the 63 newly discovered and established loci represented on Metabochip.\\n']\n",
      "['T4', 'MinorSeg 13307 13579', 'Using simulations, based on parameter estimates from this mixture model, we estimated that 488 (95% confidence interval (CI) 456–521) of the independent replication SNPs, in addition to the 63 newly discovered and established loci, are associated with T2D susceptibility. \\n']\n",
      "['T2', 'AccurateSeg 10173 10447', 'Assuming a T2D population prevalence of 8%, these 63 newly discovered and established autosomal loci together account for 5.7% of variance in disease susceptibility, as calculated by transforming dichotomous disease risk onto a continuous liability scale20 (Online Methods).\\n']\n",
      "['T1', 'AccurateSeg 10173 10447', 'Assuming a T2D population prevalence of 8%, these 63 newly discovered and established autosomal loci together account for 5.7% of variance in disease susceptibility, as calculated by transforming dichotomous disease risk onto a continuous liability scale20 (Online Methods).\\n']\n",
      "['T3', 'AccurateSeg 10667 10977', 'Figure 1 presents the distribution of Z-scores from Stage 2, aligned to the risk allele from Stage 1, at a subset of 3,412 independent (CEU r2 < 0.05) T2D replication variants that excludes lead SNPs and possible proxies (CEU r2 ≥ 0.1) at the 63 newly discovered and established loci represented on Metabochip.\\n']\n",
      "['T4', 'AccurateSeg 12787 13059', 'Using simulations, based on parameter estimates from this mixture model, we estimated that 488 (95% confidence interval (CI) 456–521) of the independent replication SNPs, in addition to the 63 newly discovered and established loci, are associated with T2D susceptibility. \\n']\n",
      "['T1', 'AccurateSeg 9801 10075', 'Assuming a T2D population prevalence of 8%, these 63 newly discovered and established autosomal loci together account for 5.7% of variance in disease susceptibility, as calculated by transforming dichotomous disease risk onto a continuous liability scale20 (Online Methods).\\n']\n",
      "['T4', 'AccurateSeg 38381 38652', 'While such a model poses challenges for accumulating genome-wide significant evidence of association at a specific variant, it does suggest that genetic profiling based on the entirety of sequence variation has the potential to provide useful risk stratification for T2D.\\n']\n",
      "['T2', 'MajorEnt 1367 1486', 'Large-scale association analysis provides insights into the genetic architecture and pathophysiology of type 2 diabetes\\n']\n",
      "['T1', 'AccurateSeg 16291 16565', 'We detected two loci at which multiple independent association signals attained genome-wide significance: KCNQ1 (rs163184, P = 1.2 × 10−11; rs231361, P = 1.2 × 10−9; CEU r2 = 0.01) and CDKN2A/B (rs10811661, P = 3.7 × 10−27; rs944801, P = 2.4 × 10−9; CEU r2 = 0.01) (Fig. 2).\\n']\n",
      "['T3', 'AccurateSeg 16566 16822', 'Both signals at KCNQ1 have previously been reported in East Asian and European populations4,24. However, the secondary signal at CDKN2A/B, which maps to the non-coding CDKN2B-AS1 (ANRIL) transcript, has not previously been implicated in T2D susceptibility.\\n']\n",
      "['T1', 'MinorEnt 1389 1508', 'Large-scale association analysis provides insights into the genetic architecture and pathophysiology of type 2 diabetes\\n']\n",
      "['T1', 'AccurateSeg 26914 27223', 'The risk allele at ANK1 has features (nominally significant reduction in HOMA-B) indicating a primary effect on beta-cell function, whereas those at GRB14 and AKNRD55 are characteristic of loci acting primarily through insulin resistance (increased HOMA-IR) (Supplementary Fig. 10 and Supplementary Table 10).\\n']\n",
      "['T1', 'AccurateEnt 2456 2701', 'To extend understanding of the genetic architecture and molecular basis of type 2 diabetes (T2D), we conducted a meta-analysis of genetic variants on the Metabochip involving 34,840 cases and 114,981 controls, overwhelmingly of European descent.\\n']\n",
      "['T2', 'AccurateSeg 26629 26841', 'To gain a more complete picture of patterns of trait overlap, we first assessed the effect of T2D risk alleles on glycemic traits in European-descent meta- analyses from the MAGIC Investigators (Online Methods). \\n']\n",
      "['T2', 'MajorEnt 948 1067', 'Large-scale association analysis provides insights into the genetic architecture and pathophysiology of type 2 diabetes\\n']\n",
      "['T3', 'MajorSection 28416 28423', 'Finally\\n']\n",
      "['T1', 'MajorSeg 3532 3835', 'The T2D-nominated component of Metabochip comprises 21,774 variants, including 5,057 “replication” SNPs that capture the strongest, independent (CEU r2< 0.2) autosomal association signals from the GWAS meta-analysis conducted by the DIAbetes Genetics Replication and Meta-analysis (DIAGRAM) Consortium. \\n']\n",
      "['T2', 'MajorSeg 3835 4109', 'This genome-wide meta-analysis (“DIAGRAMv3”) includes data from 12,171 cases and 56,862 controls of European descent imputed up to 2.5 million autosomal SNPs, and augments the previously published “DIAGRAMv2” meta-analysis4 with four additional GWAS (Supplementary Table 1).\\n']\n",
      "['T1', 'AccurateEnt 778 855', 'A PGC1α-dependent myokine that drives browning of white fat and thermogenesis\\n']\n",
      "['T2', 'AccurateEnt 778 855', 'A PGC1α-dependent myokine that drives browning of white fat and thermogenesis\\n']\n",
      "['T1', 'AccurateSection 32885 32892', 'METHODS\\n']\n",
      "['T2', 'AccurateEnt 998 1075', 'A PGC1α-dependent myokine that drives browning of white fat and thermogenesis\\n']\n",
      "['T1', 'AccurateEnt 1344 1421', 'A PGC1α-dependent myokine that drives browning of white fat and thermogenesis\\n']\n",
      "['T1', 'AccurateSeg 1600 1940', 'Exercise benefits a variety of organ systems in mammals, and some of the best- recognized effects of exercise on muscle are mediated by the transcriptional coactivator PGC1α Here we show that PGC1α expression in muscle stimulates an increase in expression of Fndc5, a membrane protein that is cleaved and secreted as a new hormone, irisin. \\n']\n",
      "['T2', 'AccurateEnt 2440 2517', 'A PGC1α-dependent myokine that drives browning of white fat and thermogenesis\\n']\n",
      "['T2', 'AccurateSection 20671 20714', 'Irisin is present in mouse and human plasma\\n']\n",
      "['T1', 'AccurateEnt 1093 1170', 'A PGC1α-dependent myokine that drives browning of white fat and thermogenesis\\n']\n",
      "['T1', 'AccurateEnt 1754 1831', 'A PGC1α-dependent myokine that drives browning of white fat and thermogenesis\\n']\n",
      "['T2', 'AccurateEnt 1379 1456', 'A PGC1α-dependent myokine that drives browning of white fat and thermogenesis\\n']\n",
      "['T1', 'AccurateEnt 1082 1190', 'Common variants in MS4A4/MS4A6E, CD2uAP, CD33, and EPHA1 are associated with late-onset Alzheimer’s disease\\n']\n",
      "['T1', 'AccurateEnt 1082 1190', 'Common variants in MS4A4/MS4A6E, CD2uAP, CD33, and EPHA1 are associated with late-onset Alzheimer’s disease\\n']\n",
      "['T1', 'AccurateEnt 2016 2124', 'Common variants in MS4A4/MS4A6E, CD2uAP, CD33, and EPHA1 are associated with late-onset Alzheimer’s disease\\n']\n",
      "['T1', 'AccurateSeg 19295 19461', 'With our findings and those by Hollingsworth et al.12, there are now ten LOAD susceptibility loci (APOE, CR1, CLU, PICALM, BIN1, EPHA1, MS4A, CD33, CD2AP, and ABCA7).\\n']\n",
      "['T2', 'AccurateEnt 1270 1378', 'Common variants in MS4A4/MS4A6E, CD2uAP, CD33, and EPHA1 are associated with late-onset Alzheimer’s disease\\n']\n",
      "['T1', 'AccurateEnt 1270 1378', 'Common variants in MS4A4/MS4A6E, CD2uAP, CD33, and EPHA1 are associated with late-onset Alzheimer’s disease\\n']\n",
      "['T1', 'AccurateSeg 5415 5584', 'Because Hollingworth et al.12 identified SNPs at ABCA7 as a novel LOAD locus, we included ABCA7 region SNPs in Stage 2 and provided the results to Hollingworth et al.12.\\n']\n",
      "['T1', 'AccurateSeg 17860 18026', 'With our findings and those by Hollingsworth et al.12, there are now ten LOAD susceptibility loci (APOE, CR1, CLU, PICALM, BIN1, EPHA1, MS4A, CD33, CD2AP, and ABCA7).\\n']\n",
      "['T1', 'AccurateSeg 18290 18455', 'With our findings and those by Hollingsworth et al.12, there are now ten LOAD susceptibility loci (APOE, CR1, CLU, PICALM, BIN1, EPHA1, MS4A, CD33, CD2AP, and ABCA7)\\n']\n",
      "['T2', 'AccurateSeg 20469 20635', 'With our findings and those by Hollingsworth et al.12, there are now ten LOAD susceptibility loci (APOE, CR1, CLU, PICALM, BIN1, EPHA1, MS4A, CD33, CD2AP, and ABCA7).\\n']\n",
      "['T2', 'AccurateEnt 2419 2527', 'Common variants in MS4A4/MS4A6E, CD2uAP, CD33, and EPHA1 are associated with late-onset Alzheimer’s disease\\n']\n",
      "['T1', 'AccurateEnt 568 676', 'Common variants in MS4A4/MS4A6E, CD2uAP, CD33, and EPHA1 are associated with late-onset Alzheimer’s disease\\n']\n",
      "['T1', 'AccurateSeg 18429 18595', 'With our findings and those by Hollingsworth et al.12, there are now ten LOAD susceptibility loci (APOE, CR1, CLU, PICALM, BIN1, EPHA1, MS4A, CD33, CD2AP, and ABCA7).\\n']\n",
      "['T1', 'AccurateSeg 838 946', 'Common variants in MS4A4/MS4A6E, CD2uAP, CD33, and EPHA1 are associated with late-onset Alzheimer’s disease\\n']\n",
      "['T2', 'MajorEnt 397 505', 'Common variants in MS4A4/MS4A6E, CD2uAP, CD33, and EPHA1 are associated with late-onset Alzheimer’s disease\\n']\n",
      "['T1', 'AccurateSeg 961 1069', 'Common variants in MS4A4/MS4A6E, CD2uAP, CD33, and EPHA1 are associated with late-onset Alzheimer’s disease\\n']\n",
      "['T2', 'AccurateSeg 18407 19017', 'Examining the amount of genetic effect attributable to these candidate genes, the most strongly associated SNPs at each locus other than APOE demonstrated population attributable fractions (PAFs) between 2.72–5.97% (Supplemental Table 9), with a cumulative PAF for non-APOE loci estimated to be as much as 35%; however, these estimates may vary widely between studies22, and the actual effect sizes are likely to be much smaller than those estimated here because of the ‘winner’s curse’. Also the results do not account for interaction among loci, and are not derived from appropriate population-based samples.\\n']\n",
      "['T2', 'AccurateEnt 2098 2196', 'Intestinal microbiota metabolism of L-carnitine, a nutrient in red meat, promotes atherosclerosis\\n']\n",
      "['T1', 'AccurateEnt 800 897', 'ntestinal microbiota metabolism of L-carnitine, a nutrient in red meat, promotes atherosclerosis\\n']\n",
      "['T1', 'AccurateEnt 1133 1230', 'ntestinal microbiota metabolism of L-carnitine, a nutrient in red meat, promotes atherosclerosis\\n']\n",
      "['T1', 'AccurateEnt 2098 2196', 'Intestinal microbiota metabolism of L-carnitine, a nutrient in red meat, promotes atherosclerosis\\n']\n",
      "['T2', 'AccurateEnt 799 897', 'Intestinal microbiota metabolism of L-carnitine, a nutrient in red meat, promotes atherosclerosis\\n']\n",
      "['T1', 'AccurateEnt 2970 3068', 'Intestinal microbiota metabolism of L-carnitine, a nutrient in red meat, promotes atherosclerosis\\n']\n",
      "['T2', 'AccurateSeg 44486 44758', 'Of note, we observed a significant increase in baseline plasma TMAO concentrations in what historically was called enterotype 2 (Prevotella) (P < 0.05), a relatively rare enterotype that previously in one study was associated with low animal fat and protein consumption18.\\n']\n",
      "['T1', 'AccurateEnt 1639 1737', 'Intestinal microbiota metabolism of L-carnitine, a nutrient in red meat, promotes atherosclerosis\\n']\n",
      "['T1', 'AccurateEnt 979 1077', 'Intestinal microbiota metabolism of L-carnitine, a nutrient in red meat, promotes atherosclerosis\\n']\n",
      "['T2', 'AccurateEnt 1639 1737', 'Intestinal microbiota metabolism of L-carnitine, a nutrient in red meat, promotes atherosclerosis\\n']\n",
      "['T1', 'AccurateEnt 1744 1842', 'Intestinal microbiota metabolism of L-carnitine, a nutrient in red meat, promotes atherosclerosis\\n']\n",
      "['T2', 'AccurateEnt 921 1019', 'Intestinal microbiota metabolism of L-carnitine, a nutrient in red meat, promotes atherosclerosis\\n']\n",
      "['T1', 'AccurateSeg 1143 1344', 'In breast cancer, gene expression analyses have defined five tumor subtypes (luminal A, luminal B, HER2-enriched, basal-like and claudin-low), each of which has unique biologic and prognostic features.\\n']\n",
      "['T2', 'AccurateSeg 3071 3316', 'Genomic studies have established four major breast cancer intrinsic subtypes (luminal A, Luminal B, HER2-enriched, basal-like) and a normal breast-like group that show significant differences in incidence, survival and response to therapy [1-3].\\n']\n",
      "['T2', 'AccurateSeg 4171 4577', 'More recently, a tumor initiating cell (TIC) genomic signature derived from CD44+/CD24-/low-sorted cells and mammospheres obtained from primary human breast tumors was found to be exclusively enriched by gene expression in the claudin-low subtype [6,7], and the expression of this CD44+/CD24-/low/claudin-low profile increased in posttreatment samples after neoadjuvant chemotherapy or hormone therapy [7].\\n']\n",
      "['T3', 'AccurateSeg 49720 50077', 'Thus, our data suggest that claudin-low tumors, compared with the other intrinsic breast tumor subtypes, are the most enriched for stem cell and/or TIC features, and on the basis of our vimentin immunofluorescence staining, it appears that these mesenchymal features are present within epithelial cells, which is a feature not seen in normal breast tissues.\\n']\n",
      "['T1', 'AccurateSeg 4171 4577', 'More recently, a tumor initiating cell (TIC) genomic signature derived from CD44+/CD24-/low-sorted cells and mammospheres obtained from primary human breast tumors was found to be exclusively enriched by gene expression in the claudin-low subtype [6,7], and the expression of this CD44+/CD24-/low/claudin-low profile increased in posttreatment samples after neoadjuvant chemotherapy or hormone therapy [7].\\n']\n",
      "['T1', 'AccurateSeg 4283 4689', 'More recently, a tumor initiating cell (TIC) genomic signature derived from CD44+/CD24-/low-sorted cells and mammospheres obtained from primary human breast tumors was found to be exclusively enriched by gene expression in the claudin-low subtype [6,7], and the expression of this CD44+/CD24-/low/claudin-low profile increased in posttreatment samples after neoadjuvant chemotherapy or hormone therapy [7].\\n']\n",
      "['T1', 'AccurateSeg 1757 1985', 'Claudin-low tumors are characterized by the low to absent expression of luminal differentiation markers, high enrichment for epithelial-to-mesenchymal transition markers, immune response genes and cancer stem cell-like features.\\n']\n",
      "['T1', 'AccurateSeg 772 1062', 'In breast cancer, gene expression analyses have defined five tumor subtypes (luminal A, luminal B, HER2-enriched, basal-like and claudin-low), each of which has unique biologic and prognostic features. Here, we comprehensively characterize the recently identified claudin-low tumor subtype.\\n']\n",
      "['T3', 'AccurateSeg 1651 1954', 'Clinically, the majority of claudin-low tumors are poor prognosis estrogen receptor (ER)-negative, progesterone receptor (PR)-negative, and epidermal growth factor receptor 2 (HER2)-negative (triple negative) invasive ductal carcinomas with a high frequency of metaplastic and medullary differentiation.\\n']\n",
      "['T4', 'AccurateSeg 31717 32078', 'In terms of prognosis, Kaplan-Meier survival analysis revealed that claudin-low tumors have a worse prognosis compared to luminal A tumors in both the UNC337 (hazard ratio [HR] of 2.83 and 5.66 for relapse-free survival [RFS] and overall survival [OS], respectively; P < 0.05) and NKI295 data sets (HR of 4.71 and 17.98 for RFS and OS, respectively; P < 0.001).\\n']\n",
      "['T5', 'AccurateSeg 32079 32261', 'Conversely, similar survival curves were observed between claudin-low tumors and the other poor prognosis subtypes such as luminal B, HER2-enriched and basal-like tumors (Figure 3b).\\n']\n",
      "['T6', 'AccurateSeg 34082 34269', 'These findings suggest that claudin-low tumors show some chemotherapy sensitivity but overall have a poor prognosis and may not be managed effectively with existing chemotherapy regimens.\\n']\n",
      "['T1', 'AccurateSeg 1107 1397', 'In breast cancer, gene expression analyses have defined five tumor subtypes (luminal A, luminal B, HER2-enriched, basal-like and claudin-low), each of which has unique biologic and prognostic features. Here, we comprehensively characterize the recently identified claudin-low tumor subtype.\\n']\n",
      "['T2', 'AccurateSection 4477 4484', 'Results\\n']\n",
      "['T1', 'AccurateSeg 31669 31829', 'Across all three databases, claudin-low tumors showed a prevalence of 7 to 14%, and were mostly ER-/PR-/HER2- (also known as triple-negative tumors, 61 to 71%).\\n']\n",
      "['T1', 'AccurateSeg 996 1286', 'In breast cancer, gene expression analyses have defined five tumor subtypes (luminal A, luminal B, HER2-enriched, basal-like and claudin-low), each of which has unique biologic and prognostic features. Here, we comprehensively characterize the recently identified claudin-low tumor subtype.\\n']\n",
      "['T1', 'AccurateSeg 22477 22782', 'Second, we observed that claudin- low tumors compared to the other tumor subtypes, except for the normal breast- like group, showed the highest mRNA expression of ALDH1A1, which has recently been proposed to be another marker of breast stem cells and TICs [34] but also noted in stromal cells [24,34,35]. \\n']\n",
      "['T3', 'AccurateSeg 23361 23555', 'To further explore the potential enrichment for breast stem cells and TIC features, we evaluated the expression of three breast stem cell-like signatures [7,15,18] across the different subtypes.\\n']\n",
      "['T4', 'AccurateSeg 58182 58396', 'In this article, we characterize an important new disease group, namely the claudin-low subtype of breast cancer, and show that these tumors have a poor prognosis and features of mesenchymal and mammary stem cells.\\n']\n",
      "['T2', 'AccurateSeg 48894 49200', 'Consistent with this finding, we observed a high mRNA expression of known transcriptional repressors of E-cadherin such as SNAI1, SNAI2, TWIST1, TWIST2, ZEB1 and ZEB2, and other EMT-inducing factors such as hypoxia-inducible factor-1a in claudin-low tumors [31] (Figure 1b, Figure S2 in Additional file 1).\\n']\n",
      "['T1', 'AccurateSeg 25222 25686', 'These nine cell lines (MDA-MB-435, MDA-MB-436, Hs578T, BT549, MDA-MB-231, MDA-MB-157, SUM1315MO2, SUM159PT and HBL100) showed a gene expression pattern similar to the claudin-low tumor subtype with the lowest expression of genes involved in epithelial cell-cell adhesion (i.e., E-cadherin and claudins 3, 4 and 7), luminal differentiation (i.e., CD24, EpCAM) and high values for the CD44/CD24 and CD49f/EpCAM mRNA ratios (Figure S4, Table S3 in Additional file 1).\\n']\n",
      "['T1', 'AccurateSeg 1968 2196', 'Claudin-low tumors are characterized by the low to absent expression of luminal differentiation markers, high enrichment for epithelial-to-mesenchymal transition markers, immune response genes and cancer stem cell-like features.\\n']\n",
      "['T1', 'AccurateSeg 5897 6157', 'All microarray and patient clinical data are available in the University of North Carolina (UNC) Microarray Database [10] and have been deposited in the Gene Expression Omnibus (GEO) under the accession number GEO:GSE18229 (referred to here as the UNC337 set).\\n']\n",
      "['T1', 'AccurateSeg 17401 17701', 'Compared to the luminal A, luminal B, HER2-enriched, and basal-like subtypes, claudin-low tumors showed inconsistent expression of basal keratins (keratins 5, 14 and 17) and low expression of HER2 and luminal markers such as ER, PR, GATA3, keratins 18 and 19 and the luminal gene cluster (Figure 1a).\\n']\n",
      "['T1', 'AccurateSeg 3974 4380', 'More recently, a tumor initiating cell (TIC) genomic signature derived from CD44+/CD24-/low-sorted cells and mammospheres obtained from primary human breast tumors was found to be exclusively enriched by gene expression in the claudin-low subtype [6,7], and the expression of this CD44+/CD24-/low/claudin-low profile increased in posttreatment samples after neoadjuvant chemotherapy or hormone therapy [7].\\n']\n",
      "['T2', 'AccurateSeg 2329 2632', 'Clinically, the majority of claudin-low tumors are poor prognosis estrogen receptor (ER)-negative, progesterone receptor (PR)-negative, and epidermal growth factor receptor 2 (HER2)-negative (triple negative) invasive ductal carcinomas with a high frequency of metaplastic and medullary differentiation.\\n']\n",
      "['T1', 'AccurateEnt 627 722', 'Phenotypic and molecular characterization of the claudin-low intrinsic subtype of breast cancer\\n']\n",
      "['T1', 'AccurateEnt 1322 1417', 'Phenotypic and molecular characterization of the claudin-low intrinsic subtype of breast cancer\\n']\n",
      "['T1', 'AccurateSeg 5877 6137', 'All microarray and patient clinical data are available in the University of North Carolina (UNC) Microarray Database [10] and have been deposited in the Gene Expression Omnibus (GEO) under the accession number GEO:GSE18229 (referred to here as the UNC337 set).\\n']\n",
      "['T2', 'AccurateSeg 5027 5287', 'All microarray and patient clinical data are available in the University of North Carolina (UNC) Microarray Database [10] and have been deposited in the Gene Expression Omnibus (GEO) under the accession number GEO:GSE18229 (referred to here as the UNC337 set).\\n']\n",
      "['T1', 'AccurateSeg 6194 6454', 'All microarray and patient clinical data are available in the University of North Carolina (UNC) Microarray Database [10] and have been deposited in the Gene Expression Omnibus (GEO) under the accession number GEO:GSE18229 (referred to here as the UNC337 set).\\n']\n",
      "['T1', 'AccurateSeg 32957 32995', 'HIV replication is inefficient in DCs.\\n']\n",
      "['T2', 'AccurateSeg 32996 33121', 'IFN stimulation in pDCs appears to be triggered mostly by endocytosed virus, whose gRNA is recognized by TLR7 in endosomes37.\\n']\n",
      "['T2', 'MajorSeg 32838 32956', 'We did not examine whether TREX1 affects IFN production by pDCs, the major source of Type I IFNs during HIV infection.\\n']\n",
      "['T1', 'MajorEnt 1703 1779', 'The cytosolic exonuclease TREX1 inhibits the innate immune response to HIV-1\\n']\n",
      "['T3', 'AccurateSection 15813 15872', 'TREX1 metabolizes HIV RT products to suppress IFN induction\\n']\n",
      "['T1', 'AccurateSection 26543 26583', 'HIV DNA signals via STING, TBK1 and IRF3\\n']\n",
      "['T3', 'AccurateSection 14701 14760', 'TREX1 metabolizes HIV RT products to suppress IFN induction\\n']\n",
      "['T1', 'AccurateSeg 2048 2274', 'In Trex1−/− mouse cells and human CD4+ T cells and macrophages in which TREX1 was inhibited by RNA interference, cytosolic HIV DNA accumulated, and HIV infection induced type I IFN that inhibited HIV replication and spreading.\\n']\n",
      "['T1', 'AccurateSection 16630 16689', 'TREX1 metabolizes HIV RT products to suppress IFN induction\\n']\n",
      "['T1', 'AccurateSeg 1605 1858', 'TREX1 bound to cytosolic HIV DNA and digested excess HIV DNA that would otherwise activate IFN expression via a TBK1, STING and IRF3 dependent pathway. HIV-stimulated IFN production in cells deficient in TREX1 did not involve known nucleic acid sensors.\\n']\n",
      "['T2', 'AccurateSection 3678 3726', 'TREX1 inhibits IFN production in response to HIV\\n']\n",
      "['T2', 'AccurateSeg 1539 1690', 'TREX1 bound to cytosolic HIV DNA and digested excess HIV DNA that would otherwise activate IFN expression via a TBK1, STING and IRF3 dependent pathway.\\n']\n",
      "['T2', 'AccurateSection 5197 5245', 'TREX1 inhibits IFN production in response to HIV\\n']\n",
      "['T1', 'AccurateEnt 1001 1077', 'The cytosolic exonuclease TREX1 inhibits the innate immune response to HIV-1\\n']\n",
      "['T1', 'AccurateEnt 984 1060', 'The cytosolic exonuclease TREX1 inhibits the innate immune response to HIV-1\\n']\n",
      "['T1', 'AccurateSeg 1524 1750', 'In Trex1−/− mouse cells and human CD4+ T cells and macrophages in which TREX1 was inhibited by RNA interference, cytosolic HIV DNA accumulated, and HIV infection induced type I IFN that inhibited HIV replication and spreading.\\n']\n",
      "['T1', 'MajorEnt 1671 1747', 'The cytosolic exonuclease TREX1 inhibits the innate immune response to HIV-1\\n']\n",
      "['T1', 'AccurateSeg 31762 31991', 'Productive HIV infection of macrophages and T cells, however, involves viral membrane fusion with the cell membrane and direct uncoating of the viral capsid into the cytosol, bypassing the endosomal compartment and TLR signaling.\\n']\n",
      "['T2', 'AccurateSeg 32876 33016', 'Some HIV DNA accumulated in the cytoplasm of HIV-infected cells, even when TREX1 is normally expressed, but did not activate IFN expression.\\n']\n",
      "['T1', 'AccurateSeg 1325 1551', 'In Trex1−/− mouse cells and human CD4+ T cells and macrophages in which TREX1 was inhibited by RNA interference, cytosolic HIV DNA accumulated, and HIV infection induced type I IFN that inhibited HIV replication and spreading.\\n']\n",
      "['T2', 'AccurateSeg 38735 38836', 'TREX1 recognizes ssDNA or dsDNA with single strand overhangs – the kind of DNA in failed RT products.\\n']\n",
      "['T4', 'AccurateSeg 38837 39008', 'TREX1 might bind to HIV DNA nonspecifically in the cytosol, but as an exonuclease can only efficiently digest HIV DNA that contains broken ends or single strand overhangs.\\n']\n",
      "['T1', 'AccurateSeg 3174 3397', 'TREX1 binds to transfected immunostimulatory DNA (ISD), and Trex1−/− cells accumulate cytoplasmic DNA derived from endogenous retroelements, which activates interferon regulatory factor 3 (IRF3)-dependent IFN expression6,7.\\n']\n",
      "['T2', 'AccurateSeg 2960 3174', 'TREX1 mutations are associated with inflammatory and autoimmune diseases, including Aicardi- Goutieres syndrome, chilblain lupus, and systemic lupus erythematosus (SLE), some of which have increased type I IFN3-5. \\n']\n",
      "['T1', 'AccurateEnt 1833 1909', 'The cytosolic exonuclease TREX1 inhibits the innate immune response to HIV-1\\n']\n",
      "['T2', 'AccurateSeg 2145 2209', 'The cytosolic exonuclease TREX1 suppressed IFN triggered by HIV.\\n']\n",
      "['T3', 'AccurateSeg 2437 2588', 'TREX1 bound to cytosolic HIV DNA and digested excess HIV DNA that would otherwise activate IFN expression via a TBK1, STING and IRF3 dependent pathway.\\n']\n",
      "['T1', 'AccurateSeg 2400 2501', 'Few copies of HIV DNA integrate, leaving behind HIV DNA in the cytosol to be cleared by host enzymes.\\n']\n",
      "['T1', 'AccurateSeg 5593 5914', 'Nevirapine, which inhibits HIV reverse transcription, but not the IN inhibitor raltegravir, which acts after HIV DNA synthesis, inhibited IFN-β and IL-6 expression in response to HIV in Trex1−/− cells (Fig. 1e,f), suggesting that HIV reverse- transcribed DNA, rather than genomic RNA (gRNA), was triggering the response. \\n']\n",
      "['T1', 'AccurateSeg 2210 2436', 'In Trex1−/− mouse cells and human CD4+ T cells and macrophages in which TREX1 was inhibited by RNA interference, cytosolic HIV DNA accumulated, and HIV infection induced type I IFN that inhibited HIV replication and spreading.\\n']\n",
      "['T1', 'AccurateEnt 1558 1634', 'The cytosolic exonuclease TREX1 inhibits the innate immune response to HIV-1\\n']\n",
      "['T1', 'AccurateSeg 2701 2927', 'In Trex1−/− mouse cells and human CD4+ T cells and macrophages in which TREX1 was inhibited by RNA interference, cytosolic HIV DNA accumulated, and HIV infection induced type I IFN that inhibited HIV replication and spreading.\\n']\n",
      "['T1', 'AccurateEnt 640 728', 'Vpx relieves inhibition of HIV-1 infection of macrophages mediated by the SAMHD1 protein\\n']\n",
      "['T1', 'AccurateEnt 335 423', 'Vpx relieves inhibition of HIV-1 infection of macrophages mediated by the SAMHD1 protein\\n']\n",
      "['T2', 'AccurateEnt 926 1014', 'Vpx relieves inhibition of HIV-1 infection of macrophages mediated by the SAMHD1 protein\\n']\n",
      "['T1', 'AccurateEnt 656 744', 'Vpx relieves inhibition of HIV-1 infection of macrophages mediated by the SAMHD1 protein\\n']\n",
      "['T2', 'AccurateSeg 18462 18849', 'Monocytes isolated from PBMCs by positive selection for CD14 using CD14 microbeads (Miltenyi Biotec Inc.) were differentiated into macrophages by culturing in DMEM supplemented with 10% heat inactivated foetal bovine serum (FBS), 10 ng/ml Macrophage-Colony Stimulating Factor (M-CSF, R&D Systems) for the first 2 days and 20 ng/ml thereafter, in 24 well plates at 3.0-3.5×105 cells/well.\\n']\n",
      "['T1', 'AccurateSeg 10357 10530', 'Importantly, the effect of Vpx was blocked when macrophages were exposed to the proteasome inhibitor MG132 prior to and during infection with Vpx-loaded SIV VLP (Figure 3b).\\n']\n",
      "['T1', 'AccurateSeg 8948 9267', 'As a negative control we selected the Vpr accessory proteins of HIV-1 and SIVmac, because they are closely related to Vpx, sharing roughly 25% and 50% amino acid identity, and can load cellular substrate proteins onto CRL4DCAF1,(21) but, unlike Vpx, Vpr does not alleviate the inhibition of lentivirus infection in MDM.\\n']\n",
      "['T2', 'AccurateSeg 9413 9501', 'SAMHD1 co-precipitated efficiently with both Vpx variants, but not with Vpr (Figure 2b).\\n']\n",
      "['T3', 'AccurateSeg 9585 9739', 'SAMHD1 was co-expressed with wild type SIVmac Vpx, a VpxQ76A variant that does not bind DCAF1, or HIV-1 Vpr that does not bind SAMHD1, in HEK 293T cells. \\n']\n",
      "['T1', 'AccurateEnt 1021 1109', 'Vpx relieves inhibition of HIV-1 infection of macrophages mediated by the SAMHD1 protein\\n']\n",
      "['T1', 'AccurateEnt 841 929', 'Vpx relieves inhibition of HIV-1 infection of macrophages mediated by the SAMHD1 protein\\n']\n",
      "['T1', 'AccurateSeg 18179 18285', 'One possible explanation is that SAMHD1-mediated restriction may require a myeloid cell-specific molecule.\\n']\n",
      "['T1', 'AccurateSeg 7759 7910', 'Specifically, we directed the assembly of three sets of protein complexes in HEK 293T cells for their subsequent proteomic characterization (Figure 2a)\\n']\n",
      "['T3', 'AccurateSeg 11459 11613', 'SAMHD1 was co-expressed with wild type SIVmac Vpx, a VpxQ76A variant that does not bind DCAF1, or HIV-1 Vpr that does not bind SAMHD1, in HEK 293T cells. \\n']\n",
      "['T4', 'AccurateSeg 19832 20052', 'SAMHD1 is expressed in HEK 293T cells, undifferentiated THP-1 cells and other non-myeloid cell types that do not possess a Vpx-sensitive mechanism restricting primate lentivirus infection (Supplementary Figures 6 and 7).\\n']\n",
      "['T1', 'AccurateSeg 2381 2584', 'Vpx relieves the inhibition of lentivirus infection in macrophages by loading SAMHD1 onto the CRL4DCAF1 E3 ubiquitin ligase, leading to highly efficient proteasome- dependent degradation of the protein. \\n']\n",
      "['T3', 'AccurateSection 57 60', 'Vpx\\n']\n",
      "['T4', 'AccurateSeg 16618 16815', 'The Vpx accessory protein of HIV-2/SIVsm lineage viruses removes this inhibition by targeting SAMHD1 for proteasome-dependent degradation via CRL4DCAF1 E3 ubiquitin ligase (Supplementary Figure 5).\\n']\n",
      "['T2', 'AccurateSeg 3230 3318', 'Vpx relieves inhibition of HIV-1 infection of macrophages mediated by the SAMHD1 protein\\n']\n",
      "['T1', 'AccurateSeg 1895 2109', 'In contrast, HIV-2 and related simian immunodeficiency viruses (SIVsm/mac) transduce myeloid cells efficiently owing to their virion-associated Vpx accessory proteins, which counteract the restrictive mechanism5,6.\\n']\n",
      "['T2', 'AccurateSeg 18507 18766', 'Of note, we observed that SAMHD1 levels are depleted by Vpx in both MDM and monocyte- derived dendritic cells in a Vpx glutamine Q76 dependent manner (data not shown) implicating SAMHD1 as the key AGS protein inhibiting lentivirus infection in myeloid cells. \\n']\n",
      "['T1', 'AccurateSeg 17404 17624', 'SAMHD1 is expressed in HEK 293T cells, undifferentiated THP-1 cells and other non-myeloid cell types that do not possess a Vpx-sensitive mechanism restricting primate lentivirus infection (Supplementary Figures 6 and 7).\\n']\n",
      "['T2', 'AccurateSeg 17625 17731', 'One possible explanation is that SAMHD1-mediated restriction may require a myeloid cell-specific molecule.\\n']\n",
      "['T1', 'AccurateSeg 18926 19185', 'Of note, we observed that SAMHD1 levels are depleted by Vpx in both MDM and monocyte- derived dendritic cells in a Vpx glutamine Q76 dependent manner (data not shown) implicating SAMHD1 as the key AGS protein inhibiting lentivirus infection in myeloid cells. \\n']\n",
      "['T2', 'AccurateSeg 18377 18597', 'SAMHD1 is expressed in HEK 293T cells, undifferentiated THP-1 cells and other non-myeloid cell types that do not possess a Vpx-sensitive mechanism restricting primate lentivirus infection (Supplementary Figures 6 and 7).\\n']\n",
      "['T4', 'AccurateSeg 18896 19122', 'Whether SAMHD1 functions autonomously, what are the other key components of a putative SAMHD1 pathway, and what are the salient features of SAMHD1 that endow this protein with anti-HIV activity, requires further investigation.\\n']\n",
      "['T5', 'AccurateSeg 17970 18087', 'One unanswered question is how myeloid cell specificity of SAMHD1-mediated inhibition of HIV-1 infection is achieved?\\n']\n",
      "['T6', 'AccurateSeg 15178 15326', 'This evidence clearly demonstrates that SAMHD1 inhibits MDM infection by HIV-1 and other primate lentiviruses by disrupting synthesis of viral cDNA.\\n']\n",
      "['T2', 'AccurateSeg 15802 15933', 'In summary, our results show that SAMHD1 interferes with HIV infection of macrophages by preventing efficient viral cDNA synthesis.\\n']\n",
      "['T1', 'AccurateSeg 15934 16131', 'The Vpx accessory protein of HIV-2/SIVsm lineage viruses removes this inhibition by targeting SAMHD1 for proteasome-dependent degradation via CRL4DCAF1 E3 ubiquitin ligase (Supplementary Figure 5).\\n']\n",
      "['T4', 'AccurateSeg 16547 16749', 'Our discovery that the SAMHD1 protein inhibits HIV infection in macrophages unveils a novel and unexpected link between unconventional cell-intrinsic innate immune mechanisms and the anti-viral defence.\\n']\n",
      "['T5', 'AccurateSeg 14130 14328', 'Specifically, it can be clearly seen that the “early” (“E”) vs. “late” (“L”) cDNA PCR readouts showed selective repression of “late” reverse transcription products at high but not low SAMHD1 levels.\\n']\n",
      "['T1', 'AccurateSeg 6126 6351', 'All protein complexes were purified by two sequential immunoprecipitations of epitope tags, under native conditions, and their subunit composition was analyzed by Multidimensional Protein Identification Technology (MudPIT)18.\\n']\n",
      "['T1', 'AccurateSeg 659 747', 'Vpx relieves inhibition of HIV-1 infection of macrophages mediated by the SAMHD1 protein\\n']\n",
      "['T3', 'AccurateSeg 15989 16145', 'Since HIV-2 and related simian viruses have evolved the Vpx function to counteract SAMHD1, they are able to infect macrophages much more readily than HIV-1.\\n']\n",
      "['T4', 'AccurateSeg 8571 8715', 'Human epitope-tagged SAMHD1 was co-expressed with Vpx, or Vpr, in HEK 293T cells and Vpx/Vpr immune complexes were analyzed by Western blotting.\\n']\n",
      "['T5', 'AccurateSeg 8716 8804', 'SAMHD1 co-precipitated efficiently with both Vpx variants, but not with Vpr (Figure 2b).\\n']\n",
      "['T6', 'AccurateSeg 8888 9042', 'SAMHD1 was co-expressed with wild type SIVmac Vpx, a VpxQ76A variant that does not bind DCAF1, or HIV-1 Vpr that does not bind SAMHD1, in HEK 293T cells. \\n']\n",
      "['T7', 'AccurateSeg 9327 9449', 'We conclude that Vpx specifically binds to, and recruits SAMHD1 to the DCAF1-DDB1 module of CRL4DCAF1 E3 ubiquitin ligase.\\n']\n",
      "['T2', 'AccurateSeg 1554 1757', 'Vpx relieves the inhibition of lentivirus infection in macrophages by loading SAMHD1 onto the CRL4DCAF1 E3 ubiquitin ligase, leading to highly efficient proteasome- dependent degradation of the protein. \\n']\n",
      "['T1', 'AccurateSeg 1963 2051', 'Vpx relieves inhibition of HIV-1 infection of macrophages mediated by the SAMHD1 protein\\n']\n",
      "['T2', 'AccurateSeg 2858 3061', 'Vpx relieves the inhibition of lentivirus infection in macrophages by loading SAMHD1 onto the CRL4DCAF1 E3 ubiquitin ligase, leading to highly efficient proteasome- dependent degradation of the protein. \\n']\n",
      "['T4', 'AccurateSection 18 21', 'Vpx\\n']\n",
      "['T5', 'AccurateSeg 17095 17292', 'The Vpx accessory protein of HIV-2/SIVsm lineage viruses removes this inhibition by targeting SAMHD1 for proteasome-dependent degradation via CRL4DCAF1 E3 ubiquitin ligase (Supplementary Figure 5).\\n']\n",
      "['T1', 'AccurateEnt 738 818', 'Transcriptomic Analysis of Autistic Brain Reveals Convergent Molecular Pathology\\n']\n",
      "['T1', 'AccurateEnt 536 616', 'Transcriptomic Analysis of Autistic Brain Reveals Convergent Molecular Pathology\\n']\n",
      "['T1', 'MajorEnt 1830 1910', 'Transcriptomic Analysis of Autistic Brain Reveals Convergent Molecular Pathology\\n']\n",
      "['T1', 'AccurateEnt 699 779', 'Transcriptomic Analysis of Autistic Brain Reveals Convergent Molecular Pathology\\n']\n",
      "['T1', 'AccurateEnt 536 616', 'Transcriptomic Analysis of Autistic Brain Reveals Convergent Molecular Pathology\\n']\n",
      "['T1', 'AccurateEnt 2452 2599', 'Here, we demonstrate consistent differences in transcriptome organization between autistic and normal brain by gene co-expression network analysis.\\n']\n",
      "['T2', 'AccurateEnt 469 549', 'Transcriptomic Analysis of Autistic Brain Reveals Convergent Molecular Pathology\\n']\n",
      "['T1', 'AccurateEnt 699 779', 'Transcriptomic Analysis of Autistic Brain Reveals Convergent Molecular Pathology\\n']\n",
      "['T2', 'AccurateSeg 21401 21710', 'These results indicate that (a) M12 consists of a set of genes that are supported by independent lines of evidence to be causally involved in ASD pathophysiology, and (b) the upregulation of immune response genes in the autistic brain observed by us and others25 has no evidence of a common genetic component.\\n']\n",
      "['T3', 'AccurateEnt 3192 3405', 'Moreover, using a published autism GWAS dataset, we show that the neuronal module is enriched for genetically associated variants, providing independent support for the causal involvement of these genes in autism.\\n']\n",
      "['T1', 'AccurateEnt 1677 1757', 'Transcriptomic Analysis of Autistic Brain Reveals Convergent Molecular Pathology\\n']\n",
      "['T1', 'AccurateSeg 20246 20597', 'As a negative control, we performed the same set-enrichment analysis using two GWAS studies for non-psychiatric disease performed on the same genotyping platform: a genome-wide association for hair color23, and a GWAS study of Warfarin maintenance dose24 finding no significant enrichment of the association signal (Figure 4b, Supplementary Figure 4).\\n']\n",
      "['T2', 'AccurateSeg 2389 2602', 'Moreover, using a published autism GWAS dataset, we show that the neuronal module is enriched for genetically associated variants, providing independent support for the causal involvement of these genes in autism.\\n']\n",
      "['T4', 'AccurateSeg 2603 2737', 'In contrast, the immune-glial module showed no enrichment for autism GWAS signals, indicating a non-genetic etiology for this process.\\n']\n",
      "['T5', 'AccurateEnt 1196 1276', 'Transcriptomic Analysis of Autistic Brain Reveals Convergent Molecular Pathology\\n']\n",
      "['T1', 'AccurateSeg 21718 21989', 'The genome-wide analysis performed here significantly extends previous findings implicating synaptic dysfunction, as well as microglial and immune dysregulation in ASD6 by providing an unbiased systematic assessment of transcriptional alterations and their genetic basis.\\n']\n",
      "['T3', 'AccurateSeg 21990 22311', 'We show that the transcriptome changes observed in ASD brain converge with GWAS data in supporting the genetic basis of synaptic and neuronal signaling dysfunction in ASD, while immune changes have a less pronounced genetic component and thus are most likely either secondary phenomena or caused by environmental factors.\\n']\n",
      "['T4', 'AccurateSeg 22312 22528', 'Since immune molecules and cells such as microglia play a role in synaptic development and function26, we speculate that the observed immune up- regulation may be related to abnormal ongoing plasticity in ASD brain. \\n']\n",
      "['T1', 'AccurateEnt 506 586', 'Transcriptomic Analysis of Autistic Brain Reveals Convergent Molecular Pathology\\n']\n",
      "['T2', 'AccurateEnt 874 954', 'Transcriptomic Analysis of Autistic Brain Reveals Convergent Molecular Pathology\\n']\n",
      "['T1', 'AccurateEnt 982 1062', 'Transcriptomic Analysis of Autistic Brain Reveals Convergent Molecular Pathology\\n']\n",
      "['T1', 'MajorEnt 982 1062', 'Transcriptomic Analysis of Autistic Brain Reveals Convergent Molecular Pathology\\n']\n",
      "['T1', 'AccurateEnt 490 570', 'Transcriptomic Analysis of Autistic Brain Reveals Convergent Molecular Pathology\\n']\n",
      "['T1', 'AccurateEnt 416 496', 'Transcriptomic Analysis of Autistic Brain Reveals Convergent Molecular Pathology\\n']\n",
      "['T1', 'AccurateEnt 490 570', 'Transcriptomic Analysis of Autistic Brain Reveals Convergent Molecular Pathology\\n']\n",
      "['T1', 'AccurateEnt 982 1062', 'Transcriptomic Analysis of Autistic Brain Reveals Convergent Molecular Pathology\\n']\n",
      "['T1', 'MajorEnt 982 1062', 'Transcriptomic Analysis of Autistic Brain Reveals Convergent Molecular Pathology\\n']\n",
      "['T1', 'AccurateEnt 1455 1535', 'Transcriptomic Analysis of Autistic Brain Reveals Convergent Molecular Pathology\\n']\n",
      "['T2', 'AccurateEnt 1477 1557', 'Transcriptomic Analysis of Autistic Brain Reveals Convergent Molecular Pathology\\n']\n",
      "['T1', 'AccurateEnt 1255 1335', 'Transcriptomic Analysis of Autistic Brain Reveals Convergent Molecular Pathology\\n']\n",
      "['T1', 'AccurateEnt 868 948', 'Transcriptomic Analysis of Autistic Brain Reveals Convergent Molecular Pathology\\n']\n",
      "['T1', 'AccurateEnt 1455 1535', 'Transcriptomic Analysis of Autistic Brain Reveals Convergent Molecular Pathology\\n']\n"
     ]
    }
   ],
   "source": [
    "present_codes = {}\n",
    "for filename in os.listdir('AccuracyNew'):\n",
    "    if '.' not in filename:\n",
    "#         res[filename] = {}\n",
    "        for file in os.listdir(f\"AccuracyNew/{filename}\"):\n",
    "            if file.endswith('ann'):\n",
    "#                 res[filename][file[:-4]] = {}\n",
    "                with open(os.path.join(\"AccuracyNew\", filename, file[:-4]+'.txt')) as f:\n",
    "                    content = f.read()\n",
    "                    idx = content.rfind('******')\n",
    "                    start_idx = idx + 9 # 6 starts above + 3 newline \n",
    "      \n",
    "                with open(os.path.join('AccuracyNew', filename, file)) as f:\n",
    "                    content = f.readlines()\n",
    "                   \n",
    "#                     line = content[1].split('\\t')[2]\n",
    "#                     res[filename][file[:-4]]['context'] = line\n",
    "#                     res[filename][file[:-4]]['support'] = []\n",
    "                    \n",
    "\n",
    "                    for c in content:\n",
    "                        l = c.split('\\t')\n",
    "                        \n",
    "                        if l[1].split()[0] in CODES:\n",
    "                            print(l)\n",
    "                            curr_code = l[1].split()[0]\n",
    "                            present_codes[curr_code] = present_codes.get(curr_code, 0) + 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "e5c72cd9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'AccurateSeg': 219,\n",
       " 'AccurateEnt': 71,\n",
       " 'MinorEnt': 3,\n",
       " 'MajorSeg': 6,\n",
       " 'MajorEnt': 10,\n",
       " 'MinorSeg': 4,\n",
       " 'MajorSection': 1,\n",
       " 'AccurateSection': 11}"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "present_codes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "ef1fec38",
   "metadata": {},
   "outputs": [],
   "source": [
    "res = {}\n",
    "\n",
    "for filename in os.listdir('AccuracyNew'):\n",
    "    if '.' not in filename:\n",
    "        res[filename] = {}\n",
    "        for file in os.listdir(f\"AccuracyNew/{filename}\"):\n",
    "            if file.endswith('ann'):\n",
    "                res[filename][file[:-4]] = {}\n",
    "                with open(os.path.join(\"AccuracyNew\", filename, file[:-4]+'.txt')) as f:\n",
    "                    content = f.read()\n",
    "                    idx = content.rfind('******')\n",
    "                    start_idx = idx + 9 # 6 starts above + 3 newline \n",
    "      \n",
    "                with open(os.path.join('AccuracyNew', filename, file)) as f:\n",
    "                    content = f.readlines()\n",
    "                   \n",
    "                    line = content[1].split('\\t')[2]\n",
    "                    res[filename][file[:-4]]['context'] = line\n",
    "                    res[filename][file[:-4]]['support'] = []\n",
    "                    \n",
    "\n",
    "                    for c in content:\n",
    "                        l = c.split('\\t')\n",
    "                        \n",
    "                        if l[1].split()[0] in CODES:\n",
    "                            start, end = int(l[1].split()[1]), int(l[1].split()[2])\n",
    "                            start, end = start - start_idx, end - start_idx\n",
    "                       \n",
    "                            res[filename][file[:-4]]['support'].append([start, end])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "539b30ea",
   "metadata": {},
   "outputs": [],
   "source": [
    "count = 0\n",
    "for x in res.values():\n",
    "    for y in x.values():\n",
    "        if len(y['support']) > 0:\n",
    "            count += 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "7ed08cb5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "217"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "6f216948",
   "metadata": {},
   "outputs": [],
   "source": [
    "import json \n",
    "\n",
    "with open('ContextAndSupports.json', 'w') as f:\n",
    "    json.dump(res, f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1fffb490",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "PT",
   "language": "python",
   "name": "pt"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
